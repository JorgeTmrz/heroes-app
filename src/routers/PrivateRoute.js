import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import PropTypes from 'prop-types'

export const PrivateRoute = ({
    isAuthenticated,
    component: Component,
    ...rest // the rest of the args passed to the component
}) => {

    const {location: {pathname: lastVisitedPage}} = rest;
    localStorage.setItem('lastPath', lastVisitedPage) // setting the las visited page on the localstorage

    return (
        <Route {...rest}
            component = {
                (props) => (
                    (isAuthenticated)
                    ? <Component {...props}/>
                    : <Redirect to = "/login" />
                )
            }
        />
    )
}

PrivateRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    Component: PropTypes.func.isRequired
}