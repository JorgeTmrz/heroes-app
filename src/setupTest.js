import "@testing-library/jest-dom";
import { configure } from "enzyme";
import { createSerializer } from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
