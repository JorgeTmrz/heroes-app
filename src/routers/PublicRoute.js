import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import { DashboardRoutes } from './DashboardRoutes'
import { PrivateRoute } from './PrivateRoute'

export const PublicRoute = ({
    isAuthenticated,
    component: Component,
    ...rest

}) => {
    return (
        <Route {...rest}
            component = {(props) => (
                (isAuthenticated)
                ? <PrivateRoute 
                    isAuthenticated = {true}
                    component = {DashboardRoutes}
                />
                : <Component/>
            )}
        />
    )
}
