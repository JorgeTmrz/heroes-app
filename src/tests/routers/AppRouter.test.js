import { mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from "../../routers/AppRouter";
import "../../setupTest";

describe("AppRouter test cases", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };

  test("should display the login if the user is not authenticated", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("should render the marvel component if its authenticated", () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: true,
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper.find(".navbar").exists()).toBe(true);
  });
});
