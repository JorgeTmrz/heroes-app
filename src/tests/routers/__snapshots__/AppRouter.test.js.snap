// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter test cases should display the login if the user is not authenticated 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Log In
  </button>
</div>
`;
